try.go

```diff
@@ -72,29 +71,90 @@
 //
 package try
 
+import (
+	"log"
+	"runtime"
+	"testing"
+)
+
 // wrapError wraps an error to ensure that we only recover from errors
 // panicked by this package.
 type wrapError struct{ error }
 
 // Unwrap primarily exists for testing purposes.
 func (e wrapError) Unwrap() error { return e.error }
 
-// Catch catches a previously panicked error and stores it into err.
-// If it successfully catches an error, it calls any provided handlers.
-func Catch(err *error, handlers ...func()) {
-	switch ex := recover().(type) {
+// Recover recovers a previously panicked error and stores it into err.
+// If it successfully recovers an error, and fn is non-nil,
+// it calls fn with the runtime frame in which the error occurred.
+//
+// Recover is a general purpose API.
+// Most use cases will be better served by Handle, HandleF, TB, or Fatal.
+func Recover(errptr *error, fn func(runtime.Frame)) {
+	r(recover(), errptr, fn)
+}
+
+// r implements recover.
+// It is a separate function from Recover to keep stack counts consistent.
+func r(recovered any, err *error, fn func(runtime.Frame)) {
+	switch ex := recovered.(type) {
 	case nil:
 		return
 	case wrapError:
 		*err = ex.error
-		for _, handler := range handlers {
-			handler()
+		if fn != nil {
+			pc := make([]uintptr, 1)
+			// 5: runtime.Callers, r, Recover/Handle/etc, the function that called defer Recover, the actual panic.
+			n := runtime.Callers(5, pc)
+			pc = pc[:n]
+			frames := runtime.CallersFrames(pc)
+			frame, _ := frames.Next()
+			fn(frame)
 		}
 	default:
 		panic(ex)
 	}
 }
 
+// Handle recovers a previously panicked error and stores it into err.
+func Handle(errptr *error) {
+	r(recover(), errptr, nil)
+}
+
+// HandleF recovers a previously panicked error and stores it into err.
+// If it successfully recovers an error, it calls fn.
+func HandleF(errptr *error, fn func()) {
+	r(recover(), errptr, func(runtime.Frame) { fn() })
+}
+
+// TB recovers any panicked errors from this package and calls tb.Fatalf.
```

Comment:
It feels weird that this package should incur a dependency on `testing` and/or `log`.

What are your thoughts on:
```go
func F(f func(string, ...any)) {
	var err error
	r(recover(), &err, func(frame runtime.Frame) {
		f("%s:%d %v", frame.File, frame.Line, err)
	})
}
```

Usage can be:
```go
func TestFoo(t *testing.T) {
    defer try.F(t.Fatalf)
    // use try.E throughout your test
}
```
or
```go
func main() {
    defer try.F(log.Fatalf)
    // use try.E throughout your test
}
```